### PR TITLE
Integ test

### PR DESCRIPTION
**Pull request by zengyan-amazon**
*Issue #, if available:*

*Description of changes:* add integ test command and one test suite

```
% yarn test:jest_integ
yarn run v1.22.0
$ node ./test/jest_integ_test.js
  console.log
    Using esRootDir: /Users/zengyan/kibana-security-workspace/kibana/plugins/opendistro_security/.es/elasticsearch-7.8.0

      at startElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:53:13)

  console.log
    Starting Elasticsearch...

      at startElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:56:11)

  console.log
    Elasticsearch pid: 99403

      at startElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:59:11)

  console.log
    Waiting for Elasticsearch to start

      at startElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:64:11)

  console.log
    pinging ES

      at startElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:70:15)

  console.log
    pinging ES

      at startElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:70:15)

  console.log
    Started Elasticsearch

      at Object.<anonymous> (plugins/opendistro_security/test/jest_integration/integration.test.ts:30:13)

 PASS  test/jest_integration/integration.test.ts (50.15s)
  start kibana server
    ✓ can access login page without credentials (37ms)
    ✓ call authinfo API as admin (275ms)
    ✓ call authinfo API without credentials (4ms)

Test Suites: 1 passed, 1 total
Tests:       3 passed, 3 total
Snapshots:   0 total
Time:        51.081s
Ran all test suites.
  console.log
    Stopping Elasticsearch

      at stopElasticsearch (plugins/opendistro_security/test/es/elasticsearch_helper.ts:94:11)

✨  Done in 57.30s.
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/es/elasticsearch_helper.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import * as fs from 'fs';
```
Comment (yuxitang-amzn) [inline test/es/elasticsearch_helper.ts:16]: Out of curiosity, why need to import this way?

File: test/es/elasticsearch_helper.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+export function downloadElasticsearch(): string {
+  if (fs.existsSync(ES_INSTALL_DIR)) {
+    childProcess.execSync(`rm -rf .es`, { cwd: PLUGIN_ROOT_DIR });
+  }
+  fs.mkdirSync(ES_INSTALL_DIR);
+
+  console.log(`Downloading Elasticsearch...`);
+  const esDownloadURL = `https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${ELASTICSEARCH_VERSION}-darwin-x86_64.tar.gz`;
+  // https://d3g5vo6xdbdb9a.cloudfront.net/tarball/opendistro-elasticsearch/opendistroforelasticsearch-1.9.0.tar.gz`;
```
Comment (yuxitang-amzn) [inline test/es/elasticsearch_helper.ts:40]: Remove unused code.

File: test/es/elasticsearch_helper.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  let pingError;
+  console.log(`Waiting for Elasticsearch to start`);
+  const kbnServerCredentials = Buffer.from(`${KIBANA_SERVER_USER}:${KIBANA_SERVER_PASSWORD}`);
+  while (countdown > 0) {
+    countdown = countdown - 1;
+    await sleep(5000);
+    try {
+      console.log('pinging ES');
+      const response = await wreck.get('https://localhost:9200', {
+        // agent: false,
```
Comment (yuxitang-amzn) [inline test/es/elasticsearch_helper.ts:97]: Remove unused code.

File: test/jest_integration/integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+import * as kbnTestServer from '../../../../src/test_utils/kbn_server';
+import { Root } from '../../../../src/core/server/root';
+import { resolve } from 'path';
+import { describe, expect, it, beforeAll, afterAll } from '@jest/globals';
+import { sleep } from '../helper/sleep';
+import { startElasticsearch, stopElasticsearch } from '../es/elasticsearch_helper';
+import { ChildProcess } from 'child_process';
+
+describe(`start kibana server`, () => {
```
Comment (yuxitang-amzn) [inline test/jest_integration/integration.test.ts:24]: Please use the single quotes for plain text, same comments applies to other occurrence in this page.

File: test/es/elasticsearch_helper.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+const ES_INSTALL_DIR = path.resolve(PLUGIN_ROOT_DIR, '.es');
+let esRootDir: string;
+
+export function downloadElasticsearch(): string {
+  if (fs.existsSync(ES_INSTALL_DIR)) {
+    childProcess.execSync(`rm -rf .es`, { cwd: PLUGIN_ROOT_DIR });
+  }
+  fs.mkdirSync(ES_INSTALL_DIR);
+
+  console.log(`Downloading Elasticsearch...`);
```
Comment (yuxitang-amzn) [inline test/es/elasticsearch_helper.ts:38]: Please use the single quotes for plain text (non string literal), same comments applies to other occurrence in this page.

File: test/es/elasticsearch_helper.ts (hunk 1 above)
Comment (zengyan-amazon) [reply test/es/elasticsearch_helper.ts:16]: not necessary, was using `import * from` when doing testing. changed to use concrete function/types

File: test/es/elasticsearch_helper.ts (hunk 2 above)
Comment (zengyan-amazon) [reply test/es/elasticsearch_helper.ts:40]: removed

File: test/es/elasticsearch_helper.ts (hunk 3 above)
Comment (zengyan-amazon) [reply test/es/elasticsearch_helper.ts:97]: removed

File: test/jest_integration/integration.test.ts (hunk 4 above)
Comment (zengyan-amazon) [reply test/jest_integration/integration.test.ts:24]: updated to use single quotes

File: test/es/elasticsearch_helper.ts (hunk 5 above)
Comment (zengyan-amazon) [reply test/es/elasticsearch_helper.ts:38]: updated

File: test/es/elasticsearch_helper.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+let esRootDir: string;
+
+export function downloadElasticsearch(): string {
+  if (existsSync(ES_INSTALL_DIR)) {
+    execSync('rm -rf .es', { cwd: PLUGIN_ROOT_DIR });
+  }
+  mkdirSync(ES_INSTALL_DIR);
+
+  console.log('Downloading Elasticsearch...');
+  const esDownloadURL = `https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${ELASTICSEARCH_VERSION}-darwin-x86_64.tar.gz`;
```
Comment (yuxitang-amzn) [inline test/es/elasticsearch_helper.ts:39]: Would suggest define a constant for this long url.

File: test/es/elasticsearch_helper.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+export function installEsSecurityPlugin() {
+  const pluginDir = resolve(esRootDir, 'plugins');
+
+  if (!existsSync(pluginDir)) {
+    mkdirSync(pluginDir);
+  }
+  execSync(
+    `yes | bin/elasticsearch-plugin install https://d3g5vo6xdbdb9a.cloudfront.net/downloads/elasticsearch-plugins/opendistro-security/opendistro_security-${SECURITY_ES_PLUGIN_VERSION}.zip`,
```
Comment (yuxitang-amzn) [inline test/es/elasticsearch_helper.ts:57]: Would suggest define a constant for this long url.

File: test/jest_integration/integration.test.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    await stopElasticsearch(esProcess);
+  });
+
+  it('can access login page without credentials', async () => {
+    const response = await kbnTestServer.request.get(root, '/app/login');
+    expect(response.status).toEqual(200);
+  });
+
+  it('call authinfo API as admin', async () => {
+    const testUserCredentials = Buffer.from(`admin:admin`);
```
Comment (yuxitang-amzn) [inline test/jest_integration/integration.test.ts:69]: Please use single quotes for plain string.

File: test/jest_integration/integration.test.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    const response = await kbnTestServer.request
+      .get(root, '/api/v1/auth/authinfo')
+      .set('Authorization', `Basic ${testUserCredentials.toString('base64')}`);
+    expect(response.status).toEqual(200);
+  });
+
+  it('call authinfo API without credentials', async () => {
+    const testUserCredentials = Buffer.from(`admin:admin`);
```
Comment (yuxitang-amzn) [inline test/jest_integration/integration.test.ts:77]: Please use single quotes for plain string.

**Pull request merged**